ethereum/vm.py

```diff
@@ -418,6 +442,10 @@ def vm_execute(ext, msg, code):
                     return vm_exception('OOG EXTENDING MEMORY')
                 mem[s0] = s1 % 256
             elif op == 'SLOAD':
+                # EIP150: Increase the gas cost of SLOAD to 200
+                if ext.post_anti_dos_hardfork():
+                    if not eat_gas(compustate, opcodes.SLOAD_SUPPLEMENTAL_GAS):
+                        return vm_exception("OUT OF GAS")
```

Comment:
It might make sense to reduce code duplication here and refactor those blocks into a helper function (c.f. storing the extra gas values in a dict from above)